torch/distributed/nn/_functions.py

```diff
@@ -0,0 +1,248 @@
+import torch
+from torch.autograd import Function
+import torch.distributed as dist
+
+
+def broadcast(tensor, src, group=dist.group.WORLD):
```

Comment:
Are we going to provide docs/examples/tutorials for when users might want to use `torch.distributed.nn` APIs over `torch.distributed`? Of course the reason is autograd, but would be good to have a plan for tutorials/docs around what use cases this can be useful for.

Reply:
I think the use cases for autograd with dist calls are way too specific to provide some kind of general tutorial. 
Just a paragraph somewhere in the docs should be ok I guess.